src/gc/collector.cpp

```diff
@@ -273,8 +391,8 @@ void markPhase() {
     VALGRIND_DISABLE_ERROR_REPORTING;
 #endif
 
-    TraceStack stack(roots);
-    GCVisitor visitor(&stack);
+    std::shared_ptr<MarkStack> stack = std::make_shared<MarkStack>(roots);
```

Comment:
I don't think there's a good reason to use shared_ptr's here - the lifetime of the visitor should be bounded by that of the stack in all cases (and should be easily verified.)